### PR TITLE
Fix/routes

### PR DESCRIPTION
**Pull request by FullStackkDev**
Create a new redux reducer for logging in and logging out the user.
Create a protected route function for checking authentication and navigating users according to authentication.
Add a new API file to get user data from the backend.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -6,49 +6,73 @@ import SignIn from "../src/screens/signIn";
...
-    getLocalStorage("user") &&
-      !Object.keys(user).lenght &&
-      dispatch(
-        userActions.addUserAfterReload(JSON.parse(getLocalStorage("user")))
-      );
-  }, []);
   return (
     <div>
-      {authenticated && <Navbar />}
+      {isAuthenticated && <Navbar />}
```
Comment (marslan-ali) [inline src/App.js:12]: @FullStackkDev please define routes in separate file

**Review by marslan-ali — CHANGES_REQUESTED** [review]

File: src/Redux/store/authSlice.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import {
+  getLocalStorage,
+  removeLocalStorage,
+  setLocalStorage,
+} from "../../helper/localStorage";
+
+const authSlice = createSlice({
+  name: "auth",
+  initialState: {
+    token: JSON.parse(getLocalStorage("user")) || null, // Get the token from local storage
```
Comment (FullStackkDev) [inline src/Redux/store/authSlice.js]: This is wrong to store the user in a token store only the token in the key "token" or update the name.

File: src/Redux/store/userSlice.js [hunk 3]
```diff
@@ -1,21 +1,13 @@
 import { createSlice } from "@reduxjs/toolkit";
-import { removeLocalStorage, setLocalStorage } from "../../helper/localStorage";
+
```
Comment (FullStackkDev) [inline src/Redux/store/userSlice.js]: remove extra space

Comment (FullStackkDev) [reply src/Redux/store/userSlice.js]: Its prettier formatting. After all imports one line is blank.

**Pull request merged**